pkg/api/v1/pod/util.go

```diff
@@ -67,6 +69,15 @@ func VisitContainers(podSpec *v1.PodSpec, visitor ContainerVisitor) bool {
 			return false
 		}
 	}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.EphemeralContainers) {
```

Comment:
I'd probably prefer this to nest the for loop rather than exit early.  Someone could easily accidentally add something after the gate.

```
if gate enabled {
  ....
}
return true
```

Reply:
SG, Done.